src/main/scala/mesosphere/marathon/core/instance/Instance.scala

```diff
@@ -109,48 +151,53 @@ object Instance {
       Condition.Killed
     ).indexOf(_)
 
+    private def stateOrHealthChanged(instance: Instance, healthy: Option[Boolean], tasks: Iterable[Task], unreachableStrategy: UnreachableStrategy, now: Timestamp) = {
+      // compute the new instance condition
+      val condition = conditionFromTasks(tasks, now, unreachableStrategy).getOrElse(Condition.Unknown)
```

Comment:
I don't think we should use `conditionFromTasks` since the instance phase is the accumulation of the tasks statuses.